dom.go

```diff
@@ -930,6 +932,8 @@ func copyProps(src, dst Component) {
 			df.Set(sf)
 		}
 	}
+	// Overwrite src with dst
+	s.Elem().Set(d.Elem())
```

Comment:
By replacing the pointer value, it allows the replacement to propagate out of the calling function, which we need for nested components.

Reply:
I think your comment here would be a good inline comment for this line instead of the existing one.

> // Effectively set *src = *dst. This allows the replacement to propagate out of
> // the calling function, which is needed for components which themselves return
> // components (see how renderComponent works for components that return
> // components).